django_netjsonconfig/base/template.py

```diff
@@ -53,6 +64,39 @@ class AbstractTemplate(BaseConfig):
                                                 'be automatically managed behind the scenes '
                                                 'for each configuration using this template, '
                                                 'valid only for the VPN type'))
+    sharing = models.CharField(_('Sharing'),
+                               choices=SHARING_CHOICES,
+                               default='private',
+                               max_length=16,
+                               db_index=True,
+                               help_text=_('Whether to keep this template private, share it publicly, '
+                                           'share it privately using a secret key or import its '
+                                           'contents from another source'))
+    key = models.CharField(max_length=64,
+                           unique=True,
+                           default=get_random_key,
+                           validators=[key_validator],
+                           help_text=_('share template key'))
+    description = models.TextField(_('Description'),
+                                   blank=True,
+                                   null=True,
+                                   help_text=_('Enter public description of this template'))
+    notes = models.TextField(_('Notes'),
+                             blank=True,
+                             null=True,
+                             help_text=_('Enter internal notes for the administrators'))
+    default_values = JSONField(_('Default Values'),
+                               default=dict,
+                               blank=True,
+                               help_text=_('Enter some default values for the variables '
+                                           'used by this template'),
```

Comment:
`A dictionary containing the default values for the variables used by this template; these default variables will be used during schema validation.`